packages/amplication-client/src/Resource/create-resource/CreateServiceWizardForm.tsx

```diff
@@ -0,0 +1,245 @@
+    undefined
+  );
+
+  const clearGeneralError = useCallback(() => {
+    setGeneralError(undefined);
+  }, [setGeneralError]);
+
+  const history = useHistory();
+
+  const serviceSettingsFieldsInitValues = {
+    generateAdminUI: true,
+    generateGraphQL: true,
+    generateRestApi: true,
+    resourceType: "scratch",
+  };
+
+  const serviceSettingsFields: MutableRefObject<serviceSettings> = useRef(
+    serviceSettingsFieldsInitValues
+  );
+
+  const [createResourceWithEntities, { loading, data, error }] = useMutation<
+    TData
+  >(CREATE_RESOURCE_WITH_ENTITIES, {
+    update(cache, { data }) {
+      if (!data) return;
+
+      const queryData = cache.readQuery<{ resources: Array<models.Resource> }>({
+        query: GET_RESOURCES,
+      });
+      if (queryData === null) {
+        return;
+      }
+      cache.writeQuery({
+        query: GET_RESOURCES,
+        data: {
+          resources: queryData.resources.concat([
+            data.createResourceWithEntities,
+          ]),
+        },
+      });
+    },
+  });
+
+  const handleSubmit = (data: serviceSettings) => {
+    if (!data.generateGraphQL) data.generateAdminUI = false;
+    serviceSettingsFields.current = data;
+  };
+
+  const errorMessage = formatError(error) || formatError(generalError);
+
+  const handleStartFromSample = useCallback(() => {
+    trackEvent({
+      eventName: "createResourceFromSample",
+    });
+    sampleServiceResourceWithEntities.generationSettings = {
+      generateAdminUI: serviceSettingsFields.current.generateAdminUI,
+      generateGraphQL: serviceSettingsFields.current.generateGraphQL,
+      generateRestApi: serviceSettingsFields.current.generateRestApi,
+    };
+
+    createResourceWithEntities({
+      variables: { data: sampleServiceResourceWithEntities },
+    }).catch(console.error);
+  }, [createResourceWithEntities, trackEvent, serviceSettingsFields]);
+
+  const handleStartFromScratch = useCallback(() => {
+    trackEvent({
+      eventName: "createResourceFromScratch",
+    });
+
+    sampleServiceResourceWithoutEntities.generationSettings = {
+      generateAdminUI: serviceSettingsFields.current.generateAdminUI,
+      generateGraphQL: serviceSettingsFields.current.generateGraphQL,
+      generateRestApi: serviceSettingsFields.current.generateRestApi,
+    };
+
+    createResourceWithEntities({
+      variables: { data: sampleServiceResourceWithoutEntities },
+    }).catch(console.error);
+  }, [createResourceWithEntities, trackEvent, serviceSettingsFields]);
+
+  useEffect(() => {
+    if (isClicked) {
+      const currentResourceType = serviceSettingsFields.current.resourceType;
+      if (currentResourceType === "scratch") {
+        handleStartFromScratch();
+      } else if (currentResourceType === "sample") {
+        handleStartFromSample();
+      } else {
+        //error?
+      }
+    }
+  }, [handleStartFromScratch, handleStartFromSample, isClicked]);
```

Comment:
a suggestion instead of if else:
useEffect(() => {
    if (!isClicked) return;
    
    const currentResourceType = serviceSettingsFields.current.resourceType;
    const handleStart = {
      scratch: handleStartFromScratch,
      sample: handleStartFromSample
    }
    handleStart[currentResourceType]()


  }, [handleStartFromScratch, handleStartFromSample, isClicked])